### DISCUSSION TITLE
[Synology] FlexGet Password help

### DISCUSSION THREAD

**Discussion by JeremieBergeron** [Flexget/Flexget#2828] [Q&A]
Hi,
When I connect to my Synology nas and I try this command `flexget web passwd test`, it display this error: root@nas:/usr/local/flexget/var# flexget web passwd test
-ash: flexget: command not found

What can I do to change the password?

Thank you

**Comment by paulpoco** [comment]
`flexget web passwd <some_password>`
should work, note you need at least 12 characters and non repeating ones.

**Comment by paulpoco** [comment]
> Hi,
> When I connect to my Synology nas and I try this command `flexget web passwd test`, it display this error: root@nas:/usr/local/flexget/var# flexget web passwd test
> -ash: flexget: command not found
> 
> What can I do to change the password?
> 
> Thank you

Are you doing that command from inside the Flexget container?   I assume on Synology uses a Docker container for Flexget?

**Reply by JeremieBergeron** [reply]
No, I am not using docker.
It is directly install from the package centor of Synology.

**Reply by JeremieBergeron** [reply]
![image](https://user-images.githubusercontent.com/34408507/108221208-867c4b80-7105-11eb-913f-41f4b6ffc485.png)
You can see here that even if I had installed Flextget, I can not access it with python, so I can not modify the password

**Comment by JeremieBergeron** [comment]
Hi,
Would it be possible to an answer?
Thank you very much

**Reply by jbesclapez** [reply]
Did you solve this issue? I have the same one :-)

**Reply by JeremieBergeron** [reply]
Sorry, I did not found a solution :(

**Comment by kerryandjane** [comment]
Hey mate. Synology don't make a flexget package. It's gotta be a community made thing. This is what I would do.

1. Run the package in package manager.
2. Log into cli and find out where and how it's running:
`ps aux |grep -a flexget` (This will give you information on where the files are kept and how it's being run.)
3. Copy the whole command (ignore the first stuff, just from /volume1/ and all the way to the end.
5. Paste the command and delete the last few words probably `daemon start` or something.
6. Add `web passwd [your long tricky password]` to the end of the line. My password is something like "Redblu2021" so the line would look like:`/volume1/python /volume1/@appstore/flexget/env/bin/flexget -c config.yml --logfile flexget.log web passwd Redblu2021` You should get a prompt saying `Updated password`

You should be good to log in now.

**Reply by jbesclapez** [reply]
I will try that. Please give me some time and I will come back to you - I especially like the "long tricky password" but we all know it is the name of my cat! LOL!

**Reply by jbesclapez** [reply]
> Hey mate. Synology don't make a flexget package. It's gotta be a community made thing. This is what I would do.
> 
> 1. Run the package in package manager.
> 2. Log into cli and find out where and how it's running:
>    `ps aux |grep -a flexget` (This will give you information on where the files are kept and how it's being run.)
> 3. Copy the whole command (ignore the first stuff, just from /volume1/ and all the way to the end.
> 4. Paste the command and delete the last few words probably `daemon start` or something.
> 5. Add `web passwd [your long tricky password]` to the end of the line. My password is something like "Redblu2021" so the line would look like:`/volume1/python /volume1/@appstore/flexget/env/bin/flexget -c config.yml --logfile flexget.log web passwd Redblu2021` You should get a prompt saying `Updated password`
> 
> You should be good to log in now.

Great! Thanks a lot for your help. It worked and I got it installed.

I now have issues with how to use it. I am trying to make make it run with a simple RSS that should download for me a torrent file in a watch folder. I can not make it run, even using the most simple example, i get the error "Error loading config: Did not pass schema validation.''
I am changing the file from the gui and i copies good to the config.yml file... so gui is working fine!
The problem is more the format. I did some research and all.. But could one provide me an example of RSS that downloads teh torrent to a folder on my synology? Can you do an example with YTS Rss? Thanks

**Reply by kerryandjane** [reply]
That's a whole other issue.
That means your yml is out of whack. Could be anything from a missing " to incorrect spacing. It's finiky as F. Find the line thats erroring out, fix it or start over and go back to basics.

I'm using a different setup. I download direct to rtorrent also running in docker. It's a direct handshake.

Are you using torrents? If you're using the standard Download station you have a few options. Either download the torrent to a folder Download station is watching or (what I used to use) [DownloadStation CLI](http://downloadstation.jroene.de/) Adding `      on_output:
        for_accepted: downloadstation add -t "video/TV_Series/{{ series_name |replace(':','')}}" "{{ url }}"
` would download the torrent and send it to the Folder "/volume1/video/TV_Series/The Flash" All depends on your set up.

**Reply by kerryandjane** [reply]
Can you share your YML? share the whole thing (edit out passwords with #### or something). schema validation issues are finicky but easy to fix.

**Reply by mosquerad** [reply]
> Hey mate. Synology don't make a flexget package. It's gotta be a community made thing. This is what I would do.
> 
> 1. Run the package in package manager.
> 2. Log into cli and find out where and how it's running:
>    `ps aux |grep -a flexget` (This will give you information on where the files are kept and how it's being run.)
> 3. Copy the whole command (ignore the first stuff, just from /volume1/ and all the way to the end.
> 4. Paste the command and delete the last few words probably `daemon start` or something.
> 5. Add `web passwd [your long tricky password]` to the end of the line. My password is something like "Redblu2021" so the line would look like:`/volume1/python /volume1/@appstore/flexget/env/bin/flexget -c config.yml --logfile flexget.log web passwd Redblu2021` You should get a prompt saying `Updated password`
> 
> You should be good to log in now.

Thank you for your great step by step, works perfect for my synology!

**Reply by goweft** [reply]
thank you

**Reply by mjbandroid** [reply]
**Rant and Notification of Fix**  

But here’s the biggest question for all of you: If you can make a **GUI for logging in**, why can’t you make a **GUI for first-time setup** that holds the user’s hand **100%**—no cut, copy, or paste required?  

As my professor always said: *"Treat everyone like a noob, and you'll win every time."* But unfortunately, most programmers seem to believe you need to be a **command-line expert** just to get things working.  

Most people who buy **off-the-shelf NAS systems** want **plug-and-play**, with a guided, **hand-holding GUI**, not a **command-line interface**. Yet, gatekeeping remains strong in this industry. And that’s why **UNIX will never be more than 4% of the consumer market**.  

If you can build a **GUI for login**, you can build a **GUI for setup**—including **permissions management**. Imagine this:  

*"You are trying to access this. Please log in as an administrator and grant this user access to these folders, files, and applications."*  

But no, that would make **too much sense** in the UNIX world. Instead, we’re expected to **dig through forums, GPT, webpages, user manuals, obscure documentation, and security references—just to get applications working and talking to each other.**


For all who are still confused
New look and update


ps aux |grep -a flexget

 /volume1/@appstore/flexget/env/bin/python3 /volume1/@appstore/flexget/env/bin/flexget -c /volume1/@appdata/flexget/config.yml --logfile /volume1/@appdata/flexget/daemon.log web passwd "your new password"

**Comment by FransAlmere** [comment]
For me /volume1/python /volume1/@appstore/flexget/env/bin/flexget -c config.yml --logfile flexget.log web passwd Redblu2021
worked ass well, thanks
